ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2017 Oleg Kushnikov, BarD Software s.r.o
+
+This file is part of GanttProject, an opensource project management tool.
+
+GanttProject is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+GanttProject is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package net.sourceforge.ganttproject.gui;
+
+import net.sourceforge.ganttproject.language.GanttLanguage;
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.ResourceAssignmentMutator;
+import net.sourceforge.ganttproject.task.Task;
+
+import javax.swing.table.AbstractTableModel;
+import java.util.*;
+
+/**
+ * @author Oleg Kushnikov
+ */
+public class ResourceAssignmentsTableModel extends AbstractTableModel {
+  enum Column {
+    ID("id", String.class),
+    NAME("taskname", String.class),
+    UNIT("unit", String.class);
+
+    private final String myCaption;
+    private final Class<?> myClass;
+
+    Column(String key, Class clazz) {
+      myCaption = GanttLanguage.getInstance().getText(key);
+      myClass = clazz;
+    }
+
+    String getCaption() {
+      return myCaption;
+    }
+
+    public Class<?> getColumnClass() {
+      return myClass;
+    }
+  }
+
+  private final List<ResourceAssignment> myAssignments;
+  private final List<ResourceAssignment> myAssignmentsToDelete = new ArrayList<>();
+  private final HumanResource myResource;
+  private final Map<Task, ResourceAssignmentMutator> myTask2MutatorMap = new HashMap<>();
+
+
+  ResourceAssignmentsTableModel(HumanResource person) {
+    myResource = person;
+    myAssignments = new ArrayList<>(Arrays.asList(person.getAssignments()));
+  }
+
+  @Override
+  public Class<?> getColumnClass(int columnIndex) {
+    return Column.values()[columnIndex].getColumnClass();
+  }
+
+  @Override
+  public int getRowCount() {
+    return myAssignments.size() + 1;
+  }
+
+  @Override
+  public boolean isCellEditable(int row, int col) {
+    return row == myAssignments.size() && Column.NAME.equals(Column.values()[col]);
```

Comment:
Right, my bad. 
Only task name is editable for new row and only load is editable for "old" rows for now.